src/gui/spatialIndicator.js

```diff
@@ -0,0 +1,370 @@
+        // if (realityEditor.gui.threejsScene.getGroundPlaneCollider()) {
+        //     objectsToCheck.push(realityEditor.gui.threejsScene.getGroundPlaneCollider());
+        // }
+        if (cachedWorldObject && objectsToCheck.length > 0) {
+            // by default, three.js raycast returns coordinates in the top-level scene coordinate system
+            let raycastIntersects = realityEditor.gui.threejsScene.getRaycastIntersects(screenX, screenY, objectsToCheck);
+            if (raycastIntersects.length > 0) {
+                // if hit a cylinder indicator, then make the cylinders expand and last longer
+                if (raycastIntersects[0].object.parent.name === 'cylinderIndicator') {
+                    console.log('spatial cursor hit a cylinder indicator in the scene!');
+                    raycastIntersects[0].object.parent.iclick++;
+                    return;
+                }
+                let groundPlaneMatrix = realityEditor.sceneGraph.getGroundPlaneNode().worldMatrix;
+                let inverseGroundPlaneMatrix = new realityEditor.gui.threejsScene.THREE.Matrix4();
+                realityEditor.gui.threejsScene.setMatrixFromArray(inverseGroundPlaneMatrix, groundPlaneMatrix);
+                inverseGroundPlaneMatrix.invert();
+                raycastIntersects[0].point.applyMatrix4(inverseGroundPlaneMatrix);
+                let trInvGroundPlaneMat = inverseGroundPlaneMatrix.clone().transpose();
+                worldIntersectPoint = {
+                    point: raycastIntersects[0].point,
+                    normalVector: raycastIntersects[0].face.normal.clone().applyMatrix4(trInvGroundPlaneMat).normalize(),
+                }
+            }
+        }
+        return worldIntersectPoint; // these are relative to the world object
+    }
+    
+    async function getMyAvatarColor() {
+        let myAvatarColor = await realityEditor.avatar.getMyAvatarColor();
+        color = `${myAvatarColor.color}`;
+        colorLighter = `${myAvatarColor.colorLighter}`;
+        finalColor[0] = {
+            color: new THREE.Color(color),
+            colorLighter: new THREE.Color(colorLighter)
+        };
+    }
+    
+    let spatialIndicatorActivated = false;
+    function handleMouseClick(e) {
+        // if (!avatarActive) return;
+        spatialIndicatorActivated = true;
+        worldIntersectPoint = getRaycastCoordinates(e.clientX, e.clientY);
+        if (worldIntersectPoint !== undefined) addSpatialIndicator();
+    }
+
+    const indicatorAxis = new THREE.Vector3(0, 1, 0);
+    const indicatorHeight = 400;
+    const indicatorName = 'cylinderIndicator';
+    const iDuration = 5;
+    const iAnimDuration = 1;
+    const iScaleFactor = 1.1;
+    let indicatorList = [];
+    
+    let innerWidth = 20;
+    let innerBottomHeight = 70;
+    let innerTopHeight = 250;
+    let innerHeightOffset = 50;
+    
+    function addSpatialIndicator() {
+        console.info('should add a cylinder to the scene');
```

Comment:
We should get rid of these logging statements before committing to avoid clogging up the console.

Reply:
Right! Will delete these logs. Thx Dan!